Review

MAsmita opened the pull request.

zahir-koradia commented on src/test/java/APITest/dataprovide.java:
```diff
@@ -0,0 +1,35 @@
+package APITest;
```
Typo in file name.

zahir-koradia commented on src/test/java/APITest/Payload.java:
```diff
@@ -0,0 +1,53 @@
+	  
+	// 1.. register successful
+	public static String register()
+		{
```
It is important to indent the code consistently as it makes it much more easy to read/write/modify the code. Most IDEs have options/shortcuts to auto indent the code. Please find a way to do it for your IDE so that you don't have to do the indentation manually.

zahir-koradia commented on src/test/java/APITest/dataprovide.java:
```diff
@@ -0,0 +1,35 @@
+	    {
+
+	   RestAssured.baseURI="https://reqres.in";
+	   String response= given().header("Content-Type","application/json")
```
Space before and after `=`

zahir-koradia commented on src/test/java/APITest/dataprovide.java:
```diff
@@ -0,0 +1,35 @@
+	             .when().post("/api/users")
+	             .then().log().all().assertThat().statusCode(201).extract().response().asString();
+             
+		 System.out.println(response);
```
Remove prints. They are useful when debugging or when writing the test, but should be removed from the final code so that it is easier to see which tests failed when tests are run in bulk.

zahir-koradia commented on src/test/java/APITest/dataprovide.java:
```diff
@@ -0,0 +1,35 @@
+
+	   RestAssured.baseURI="https://reqres.in";
+	   String response= given().header("Content-Type","application/json")
+	             .body(Payload.create_multiple_user(name, job, id))
```
Use Pojo classes instead of json payloads.

zahir-koradia commented on src/test/java/APITest/deserialisation.java:
```diff
@@ -0,0 +1,27 @@
+		           System.out.println("First name: " +(resp.getData().get(1).getFirst_name()));
+		           System.out.println("company name: "+( resp.getAd().getCompany()));
+		
+	
```
Remove empty lines. Usually the auto-indent (also called auto-format) utility in your IDE will do this for you too.

zahir-koradia commented on src/test/java/APITest/deserialisation.java:
```diff
@@ -0,0 +1,27 @@
+import io.restassured.parsing.Parser;
+import pojo.CreateUser;
+
+public class deserialisation {
```
Java follows the convention of naming classes in CapitalCase. So here the `d` should be capitalized.

zahir-koradia commented on src/test/java/APITest/register_sucess.java:
```diff
@@ -0,0 +1,55 @@
+	String response= given().spec(Utility.requestbuild()) 
+		             .body(Payload.register())
+                     .when().post("/api/register")
+                     .then().spec(Utility.responsebuild()).assertThat().body("id",equalTo(4)).extract().response().asString();
```
Avoid long lines as they are hard to read. Again something your auto-formatter will take care of.

zahir-koradia replied:
Use Pojo classes for response.

zahir-koradia commented on src/test/java/APITest/register_sucess.java:
```diff
@@ -0,0 +1,55 @@
+             System.out.println(response);
+    
+     //Register Unsuccessful
+	 String unsuccessregister =given().spec(Utility.requestbuild())
```
Each test case should be put in a separate method. Then you should call the test methods from the main method.

zahir-koradia commented on src/test/java/APITest/register_sucess.java:
```diff
@@ -0,0 +1,55 @@
+	           .when().get("/api/users")
+               .then().spec(Utility.responsebuild()).extract().response().asString();
+	            
+               JsonPath js=new JsonPath(delayedresp);
```
The next 4 lines are not doing any assertion. Either they should be doing assertions or should be removed.

MAsmita commented:
hello Zahir Bhai
i have created copy of dataprovider class named dataprovider2. i have used pojo instead of json body. i have tried few times different ways to run that test case but its not working. can u plz let me know where is the problem and where i have gone wrong.
thank you

MAsmita commented:
hello Zahir Bhai
I have resolved all the suggested changes. I have used Pojo classes where advised, changed all the class name where suggested and all the spaces before = and after. Removed unnecessary print,log(). add() statements and Jsonpath.  Wrapped up Test cases in seperate Methods.
 Can u please look into my work again and let me know.
Thank you

zahir-koradia commented on src/test/java/APITest/Dataprovider.java:
```diff
@@ -0,0 +1,24 @@
+@Test(dataProvider = "registerdetails")
+public void user_registration(String name, String job) {
+
+	RestAssured.baseURI = "https://reqres.in";
```
Since the base uri and content-type are the same across all tests may be you can use a request builder and share it across all the tests.

zahir-koradia commented on src/test/java/APITest/Dataprovider.java:
```diff
@@ -0,0 +1,24 @@
+		.then().assertThat().statusCode(201).extract().response().asString();
+ }
+
+@DataProvider(name = "registerdetails")
```
All the methods in a class should be indented.

zahir-koradia commented on src/test/java/APITest/Dataprovider2.java:
```diff
@@ -0,0 +1,26 @@
+			 {new CreatePost2("michel2", "technician2")}
+			 };
+  }
+}
```
Add a new line here at the end of the file. Your IDE should do this as a part of autoformatting.

zahir-koradia commented on src/test/java/APITest/Dataprovider.java:
```diff
@@ -0,0 +1,24 @@
+import org.testng.annotations.Test;
+import io.restassured.RestAssured;
+
+public class Dataprovider {
```
This is a bad class name. Class name should represent which api it is testing.

zahir-koradia commented on src/test/java/APITest/Dataprovider.java:
```diff
@@ -0,0 +1,24 @@
+package APITest;
```
Why are there multiple classes testing the same API?

zahir-koradia commented on src/test/java/APITest/List_users.java:
```diff
@@ -0,0 +1,64 @@
+import static org.hamcrest.Matchers.*;
+import ResponsePojo.Response_single_user;
+
+public class List_users {
```
Class names should be `CapitalCased`. That is the convention followed in Java.

zahir-koradia commented on src/test/java/APITest/List_users.java:
```diff
@@ -0,0 +1,64 @@
+  }
+}	
+		
+	
```
Remove empty lines. There should one empty line at the end of the file.

zahir-koradia commented on src/test/java/APITest/List_users.java:
```diff
@@ -0,0 +1,64 @@
+	List_users.single_resource_notfound();
+}
+		        
+public static void All_list_users() {
```
Use the @test directive to ensure testng runs the tests as a part of the suite.

zahir-koradia commented on src/test/java/APITest/List_users.java:
```diff
@@ -0,0 +1,64 @@
+	.body("data.id", hasItems(7,8));    
+  }      
+
+public static void Delete_user() {
```
File name should be `camelCased`. Please read about naming conventions here: https://www.geeksforgeeks.org/java-naming-conventions/

zahir-koradia commented on src/test/java/APITest/Payload.java:
```diff
@@ -0,0 +1,42 @@
+
+public class Payload {
+	  
+public static String register() {
```
You shouldn't need this whole file if you are using POJO clases everywhere.

zahir-koradia commented on src/test/java/APITest/serialisation1.java:
```diff
@@ -0,0 +1,42 @@
+import ResponsePojo.response_update_job;
+import pojo.CreatePost;
+
+public class Serialisation1 {
```
Method/class names should not contain 1/2 etc numbers. That usually is an indication that they haven't been named properly.

zahir-koradia commented on src/test/java/pojo/CreatePost.java:
```diff
@@ -0,0 +1,37 @@
+	public void setEmail(String email) {
+		this.email = email;
+	}
+	
```
Remove empty lines.

zahir-koradia commented on src/test/java/pojo/Data.java:
```diff
@@ -0,0 +1,41 @@
+	public void setAvatar(String avatar) {
+		this.avatar = avatar;
+	}
+	}
```
The `}` should be 0 indented.
